servlet/src/main/java/io/undertow/servlet/spec/ServletPrintWriter.java

```diff
@@ -254,45 +236,84 @@ public void write(final char[] buf, final int off, final int len) {
         write(cb);
     }
 
+    private static int writeAndFlushAscii(ServletOutputStreamImpl outputStream, ByteBuffer buffer, char[] chars, int start, int end) throws IOException {
+        assert buffer.order() == ByteOrder.BIG_ENDIAN;
+        int i = start;
+        while (i < end) {
+            final int bufferPos = buffer.position();
+            final int bufferRemaining = buffer.remaining();
+            final int sRemaining = end - i;
+            final int remaining = Math.min(sRemaining, bufferRemaining);
+            final int written = setAsciiBE(buffer, bufferPos, chars, i, remaining);
+            i += written;
+            buffer.position(bufferPos + written);
+            if (!buffer.hasRemaining()) {
+                outputStream.flushInternal();
+            }
+            // we have given up with the fast path? slow path NOW!
+            if (written < remaining) {
+                return i;
+            }
+        }
+        return i;
+    }
+
+    private static int setAsciiBE(ByteBuffer buffer, int out, char[] chars, int off, int len) {
+        final int longRounds = len >>> 3;
+        for (int i = 0; i < longRounds; i++) {
+            final long batch1 = (((long) chars[off]) << 48) |
+                    (((long) chars[off + 2]) << 32) |
+                    chars[off + 4] << 16 |
+                    chars[off + 6];
+            final long batch2 = (((long) chars[off + 1]) << 48) |
+                    (((long) chars[off + 3]) << 32) |
+                    chars[off + 5] << 16 |
+                    chars[off + 7];
+            if (((batch1 | batch2) & 0xff80_ff80_ff80_ff80L) != 0) {
+                return i << 3;
+            }
+            final long batch = (batch1 << 8) | batch2;
```

Comment:
@theRealAph I've removed 

```java
            final long maskedBatch1 = (batch1 & 0x007f_007f_007f_007fL) << 8;
            final long maskedBatch2 = batch2 & 0x007f_007f_007f_007fL;
            final long batch = maskedBatch1 | maskedBatch2;
```
because of the previous check re being  in `[0, 127]`